File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        origin_s = ''
        opposite_s = ''

        for c in s:
            if c.isalnum():
                c_lower = c.lower()
                origin_s = origin_s + c_lower
                opposite_s = c_lower + opposite_s
        
        return origin_s == opposite_s

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
#include <cctype>

class Solution {
public:
    bool isPalindrome(string s) {
        int toLower = 32;
        string from_left;
        string from_right;

        for (auto c: s) {
            if (isalnum(c)) {
                if (c >= 'A' && c <= 'Z')
                    c += toLower;
                from_left = from_left + c;
                from_right = c + from_right;
            }
        }
        return from_left == from_right;
    }
};